src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ConnectionExceptionTest.cs

```diff
@@ -18,6 +18,31 @@ public class ConnectionExceptionTest
         private const string execReaderFailedMessage = "ExecuteReader requires an open and available Connection. The connection's current state is closed.";
         private const string orderIdQuery = "select orderid from orders where orderid < 10250";
 
+        [Fact]
+        public void TestConnectionStateWithErrorClass20()
+        {
+            using TestTdsServer server = TestTdsServer.StartTestServer();
+            using SqlConnection conn = new(server.ConnectionString);
+            conn.Open();
+            SqlCommand cmd = conn.CreateCommand();
+            cmd.CommandText = "SELECT 1";
+            int result = cmd.ExecuteNonQuery();
+            Assert.Equal(-1, result);
+            Assert.Equal("Open", conn.State.ToString());
+            server.Dispose();
+            try
+            {
+                int result2 = cmd.ExecuteNonQuery();
+            }
+            catch (SqlException ex)
+            {
+                Assert.True(ex.Class >= 20);
```

Comment:
Here there is failure as SqlException returns Error class 11. inside the driver the caught exception has class 20, however on `internal static SqlException CR_AllAttemptsFailed` inside SqlUtil class the error number is set to MIN_ERROR_CLASS, 11. I believe that needs to be updated.

Reply:
Code shouldn't need modification.. your test server needs to throw appropriate error when you run query.

Reply:
Since the server is not available in this case there is nothing to be done from the server side. The error class 20 is from the inner exception. The parent exception is always 11.  I have adjusted the test to address our request correctly.